servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/NettyIoThreadFactory.java

```diff
@@ -17,20 +17,20 @@
 
 import io.servicetalk.concurrent.api.AsyncContextMap;
 import io.servicetalk.concurrent.api.AsyncContextMapHolder;
+import io.servicetalk.transport.api.IoThreadFactory;
 
 import io.netty.util.concurrent.FastThreadLocalThread;
 
-import java.util.concurrent.ThreadFactory;
 import java.util.concurrent.atomic.AtomicInteger;
 import javax.annotation.Nullable;
 
 import static java.lang.Thread.NORM_PRIORITY;
 import static java.util.Objects.requireNonNull;
 
 /**
- * Default {@link ThreadFactory} to create IO {@link Thread}s.
+ * Default {@link NettyIoThreadFactory} to create IO {@link IoThread}s.
  */
-public final class IoThreadFactory implements ThreadFactory {
+public final class NettyIoThreadFactory implements IoThreadFactory<NettyIoThreadFactory.NettyIoThread> {
```

Comment:
Even for internal modules, we should be extra cautious and maintain backward bytecode compatibility for 0.41, because some older ST version may require previous `IoThreadFactory`. Consider keeping both for now: `IoThreadFactory` and `NettyIoThreadFactory`.

Reply:
The name change is because spotbugs won't allow the new interface and the existing class to have the same name. We would need to add an exclusion rule.

Reply:
Name changed back to IoThreadFactory and spot bugs exclusion rule applied.